python/pyspark/sql/catalog.py

```diff
@@ -133,6 +133,35 @@ def listColumns(self, tableName, dbName=None):
                 isBucket=jcolumn.isBucket()))
         return columns
 
+    def tableExists(self, tableName, dbName=None):
+        """Check if the table or view with the specified name exists.
+        This can either be a temporary view or a table/view.
+
+        .. versionadded:: 3.3.0
+
+        Parameters
+        ----------
+        tableName : string
+                    name of the table to check existence
+        dbName : string, optional
+                 name of the database to check table existence in.
+                 If no database is specified, the current database is used
+
+        Returns
+        -------
+        Boolean indicating whether the table/view exist
```

Comment:
```suggestion
        bool
            Indicating whether the table/view exist
```

Reply:
per https://numpydoc.readthedocs.io/en/latest/format.html :-)

Reply:
Also updated ! Thanks for the comment